### DISCUSSION TITLE
【小游戏分享】搬运一款童年坦克游戏 最近刚玩到

### DISCUSSION THREAD

**Discussion by Buncher123** [share121/inter-knot#994] [其他]
![Title](https://github.com/user-attachments/assets/1dbe9a72-115b-4947-90c1-69d95584847f)
![CookieHelp](https://github.com/user-attachments/assets/ada8dda0-1852-44d3-9975-342a23ff8322)
![image](https://github.com/user-attachments/assets/e9749eea-a51d-4fe7-bef8-46a5ca346112)

决战坦克是一款非常好用且十分经典的平面坦克射击类游戏，同时也是大部分人童年的回忆。在这个游戏里玩家需要操控你的坦克，以保护基地为主旨，而展开一段惊心动魄的战斗征程，那么你最终能否保护好你的基地呢？这就要看你的表现如何了。

网盘自取：链接：https://pan.baidu.com/s/171BjvxENL-FotRZBPdxzwg?pwd=2024 
提取码：2024

What is new in v1.2.2 ?
 1、增加了存档、读档功能，存档是自动的；
 2、降低了游戏难度，其中：敌方布雷车的布雷频率降低到 20 %；
 3、调整了子弹速度，其中：穿甲弹在发射初期速度较慢；
 4、修正了游戏速度问题（呵呵，150 % 的速度是很爽的）；
 5、修正了奖励分数问题。
 6、处理了背景图案，普遍把亮度和对比度降低了（希望这次可以看见 Mine 
    了）。
 7、增加了 6 张地图，共 52 张地图。

What is new in v1.2.0 ?
 1、增加了一种新的加强武器（Mine）；
 2、增加了两种新的辅助选件（Shield Of Moon，Memory of Sea）；
 3、增加了五种新的宝物；
 4、增加了八种新的敌人；
 5、地图总数增加到 46 （不要使用旧的地图编辑器来编辑关卡文件）；
 6、地图的分布与难度有关，部分关卡要通过了“极困难”级别才会出现；
 7、某些地图的某些位置有隐藏宝物，打掉该砖块就会出现（隐藏宝物当然
    是最好的东西啦）；
 8、佩戴 Memory of Sea 的敌人可以获得除了“分数”以外的任何宝物，
    有时后果很严重的，要小心啦；
 9、火焰被分成了三种颜色，“红”（对敌我有损）、“橙”（仅对我方有
    损）、“黄”（仅对敌方有损）；
10、装甲上限增加到 8，子弹数量上限增加到 5 倍基数；
11、还有很多细节上的改良，比如：你可以在1.5sec内回避突然出现在你头
    顶的宝物，免得飞来横祸；敌人比你晚出现4sec，于是你可以占领有利
    位置；续关的时候，会给予额外的支持和保护。
12、更多的细节不在这里一一列举了，相信广大玩家 1 Try 就 Know 。 

对 Mine 的补充说明：
    当你坐完几次免费飞行后，你就会意识到 Mine 是种危险的武器。你有
    必要了解辅助选件跟 Mine 的关系：

    a) [ AmphiTank ], 可以在水上布 Mine ；
b) [ AntiSkid ], 可以在敌方 LV0～LV2 的 Mine 上安全行走（LV3 的
       Mine 是有彩环的）；
    c) [ Shield of Moon ], 可以安全去除敌方 LV0 的 Mine；
    d) [ Memory of Sea ], 可以免坐飞机之险，免除慢行之苦（前提是你
       有足够的装甲或者无敌，否则见不到彩云，却见到了上帝）；

    还有的是爆破弹可以引爆等于或低于自己等级的 Mine；

What is new in v1.1.1 ?
1、改正了联网传输和联网选项的两个小Bug。
2、减少了地图传输的网络负荷。
3、联网时，如果有玩家进入或退出游戏，系统会提示信息。
4、现在，在默认状态下，自动暂停游戏的功能是关闭的。你可以按[F8]切
换开关或者按[F9]看性能选项下的“自动暂停游戏”。（这样是为了在联网
主机玩家处理其他事情的时候，客户机不会自动暂停）

注：[F1]、[F4]...[F12]上都定义了某些功能，请不要用于游戏控制。
    [F1]  在线帮助；     [F4]  全屏/窗口模式切换；
    [F9]  设置选项；     [F10] 强制刷新画面；
    [F11] 隐藏/显示信息；[F12] 发送信息。

What is new in v1.1.0 ?
1、改善了联网性能。本来可以做得更好的，只是 IDirectPlay4 for DX7
功能太weak了。建议搞网络游戏的开发人员使用 DirectPlay8，MSDN 上
描述的功能很强大。（为什么我不用？为了让小游戏不要依赖太高。DX7是
标配嘛。想用的高级功能自己做好了）
2、联网时可以发送信息，按[F12]。为什么不用[Enter]？为了不给喜欢用
[Enter]作为发射键的玩家带来麻烦。毕竟这不是实时战术游戏（那是以鼠
标为主）。[F12]在[Enter]正上方两个键的距离，习惯了也很好用。
3、你可以屏蔽别人的信息，按[F9]看联网选项。（有hint）
4、你可以方便的查看自己的IP地址，按[F9]看联网选项下的[信息]按钮。

What is new in v1.0.0 ?
1、增加了联网功能，详细见《说明文档》。
2、签名的方法更改了，详细见《说明文档》。
3、增加了封面。
4、对不同级别的速度进行了调整，自己坦克调快了，敌人坦克调慢了。
5、按[Fire 1]和[Fire 2]都可以跳过开场、进入游戏、加快过场。
6、Update 了《说明文档》，更新内容有(*)标记。
7、还有好几个小地方更改了，……，又记不清了。

What is new in v0.9.3 ?
1、游戏速度可以设置，从 83% 到 150% 。
2、还好象改了写什么，记不清了。

What is new in v0.9.2 ?
1、增加了设置控制键的功能。
2、增加了存储和读取设置的功能。
3、增加了按[Esc]时退出的提问。
4、对[密集火力]在LV2、LV3过分散射的问题加以修正。
5、增加了 5幅地图。（谁能帮我画地图啊？我把MapEditor发给你）
6、对以前的20幅地图的坦克数量进行了调整。（放心吧，不是增加）
7、还好象改了写什么，记不清了。

＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝
　　　　　　　　　　　决战坦克 v1.0.0 说明文档
＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝

一、游戏特征：

  　平面坦克射击类游戏，具有４种加强型武器、２种附加能力、４个速度级
别、４个武器级别、２０种宝物，４类１６种敌人坦克，５种难度选择。敌人
具有主动进攻金鹰（基地）的智能（由难度定）。

  　能在 640Ｘ480Ｘ16 bits 下流畅运行。绿色软件，无须安装。

二、游戏目的：

  　保护金鹰（基地），消灭所有敌人坦克。

三、联网游戏：(*) 

    在封面点击 [单机游戏]，开始一个单机游戏。

    在封面点击 [联网游戏]，开始一个联网游戏。

    在点击 [联网游戏] 后，游戏弹出一个向导，用于通讯设置。如果您使用
LAN 联网，可以选择“IPX 连接”或“Internet TCP/IP 连接”；如果您使用
互联网联网，可以选择“Internet TCP/IP 连接”。

    您应该让网络速度最快的计算机做主机，选择“开始一个新游戏”，输入
游戏名称和玩家名字，点击 [完成]；然后其他（最多 3 台）计算机做客户机，
选择“加入一个己经开始的游戏”，输入主机的 IP 或域名；接着计算机会连
接主机，下载游戏列表，选择该游戏名称并且输入玩家名字，点击 [完成]。

    如果两个计算机之间通过了虚拟IP网络网关，就很可能无法得到游戏列表。
（谁想到怎样设置 DXPlay 可以解决这个问题，给我来信）

    您可以在任何时间加入主机的游戏，即使游戏已经开始。在加入游戏后，
您就可以看见主机的游戏画面，如果主机允许您控制并且坦克没有人在控制
（还没有投币或主机终止坦克的控制），您按对应的 [射击] 就可以控制该坦
克。比如：主机控制着一号坦克，您应该按二号坦克的 [射击]。

四、联网游戏时的选项：(*) 

    在联网游戏中按 [F9] 可以看到“联网”一页，主机可以控制其中的选项。
比如：您要踢走一个玩家，就按下该玩家的名字；您要禁止新玩家进入，就按下
 [新玩家] 的按钮；您要一个玩家终止或禁止对二号坦克的控制，就不选名字后
面的 [控制P2] 检查筐；您要后来的玩家都只能观看游戏，就不选 [新玩家] 后
面的 [控制P1]、[控制P2] 检查筐。这些选项在点击 [OK] 后才生效。客户机中
显示的“联网”选项跟主机的同步。

    在联网游戏中，调节客户机的“杂项”不起作用，调节客户机的“性能”不
起作用。（“影子不透明度”除外）

    还有，您可以根据网络速度调节主机的“性能”一页中的“网络速度”，网
络慢就左调，网络快就右调。最后，关闭主机的“使用半透明的屏幕指示”可以
加快游戏速度。

五、游戏控制：

    以下是默认控制按键。按 [F9] 可以打开游戏选项，在“控制”一页中修改控制
按键。

  　[Player1] 　[Player2]
　  　按键　　　　按键　　 　游戏时功能　
  　－－－－－－－－－－－－－－－－－－－
  　　 Ｗ　　　 　 ↑　　　　　　上　　　
　  　 Ｓ　　　　  ↓　　　　　　下　　　
  　　 Ａ　　　　　←　　　　　　左　　　
　　   Ｄ　　　　　→　　　　　　右　　　
  　　U或J 　　　Num4或1　　　普通射击　 
　  　I或K　 　　Num5或2　　　加强射击　　

  　在标题画面按[射击]，就开始游戏。游戏中随时按[射击]就可以加入战斗，
或继续游戏（接关）。在过场时按[射击]可以加快速度。

(*) 现在当需要签名时，在单机游戏会弹出对话筐，在联机游戏会以通讯设置
中的名字为准。签名必须是14个字符以内的英文。

(*) 关于游戏的暂停，只要按 [F9]、[F1] 或 使游戏失去焦点，游戏就自动暂
停。（注意：联网时客户机无法暂停 ）

六、功能键作用：

  　[F1]　：　在线帮助；
  　[F4]　：　全屏模式和窗口模式切换；
  　[F5]　：　显示 FPS ；
  　[F6]　：　自动跳帧开关切换；
  　[F7]　：　透明屏幕指示开关切换；
  　[F9]　：　设置选项。
  　[F10] ：　刷新画面。

七、宝物功能和屏幕指示说明：

  　请查看[在线帮助]，即在游戏中按[F1]。

八、关于背景音乐

  　MIDI太难听，MP3就太大，不如自己开个Winamp，想听什么听什么。

    有的带Vid的Winamp在换歌的时候会让全屏模式的画面闪动，请按[F10]解决。

九、关于版权

  　免费软件。作者大大保留所有权利，未经允许不得用于商业用途。

　  热情与广大软件开发爱好者交流，E-mail: Leonardo263090@21cn.com

**Comment by gaojiaobei** [comment]
好
